### ISSUE TITLE
_calculate_similarities crashes when tracker_dets_t is empty

### ISSUE THREAD

**Issue by abrayyan** [JonathonLuiten/TrackEval#56]
For some frames when there are no tracks (tracker_dets_t empty), _calculate_similarities calls either _calculate_box_ious or _calculate_mask_ious assuming both contains objects

**Comment by ThanosRoidis** [comment]
yes, this is a real problem, if you have no detections in a frame then an exception is raised when trying to calculate the similarity matrix
